### PR TITLE
feat: add custom undici instance

### LINKED ISSUE #192: Use a custom undici instance
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

👋🏻 Thank you for this awesome package! I was thinking that it could be useful to allow to pass a custom `undici` instance  in addition to or as an alternative to the custom options already available. WDYT?

### Motivation

The first 2 reasons that come to mind are:

* use a `mockedClient` in tests
* re-use an instance already initialised in the app

### Example

_No response_

### PR DESCRIPTION
**Pull request by dnlup**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Closes #192

### CODE REVIEW

File: test/custom-undici-instance.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+t.test('use a custom instance of \'undici\'', async t => {
+  t.teardown(target.close.bind(target))
+
+  await new Promise((resolve, reject) => target.listen(0, err => err ? reject(err) : resolve()))
+
+  t.test('custom Pool', t => {
+    const instance = Fastify()
+    t.teardown(instance.close.bind(instance))
+    instance.register(From, {
+      base: `http://localhost:${target.address().port}`,
```
Comment (dnlup) [inline test/custom-undici-instance.js:30]: Unless undici adds a way to get the target url, the base has two be specified in the options. Is that ok?

**Comment by dnlup** [comment]
If the changes are approved, I'll also modify the README.

File: test/custom-undici-instance.js (hunk 1 above)
Comment (mcollina) [reply test/custom-undici-instance.js:30]: It's ok for now, however we might want to change it in undici! Maybe a PR there?

**Review by mcollina — APPROVED** [review]
lgtm

File: test/custom-undici-instance.js (hunk 1 above)
Comment (dnlup) [reply test/custom-undici-instance.js:30]: Yes, that makes sense 😉

**Pull request merged**